### DISCUSSION TITLE
Resolving "implicit" dependencies

### DISCUSSION THREAD

**Discussion by mattmacdonald-m42** [garronej/denoify#88] [Q&A]
Hello!

I was experimenting with Denoify-ing a NestJS test project to see how well it would work with Deno. The default code will start you with an Express server, but requires you to add it as a dependency: `@nestjs/platform-express`. What I was running into was that when I use Denoify this dependency is not included anywhere in the output, which makes sense since it's not referenced explicitly in my code anywhere. However it stops the application from being able to run.

I was able to work around this by just explicitly setting the backing server type (or, optionally, an empty `import` statement)

E.g.
Before (valid):
```ts
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';

async function bootstrap() {
    const app = await NestFactory.create(AppModule);
    await app.listen(3001);
}
bootstrap();
```
After:
```ts
import { NestFactory } from '@nestjs/core';
import { NestExpressApplication } from '@nestjs/platform-express';
import { AppModule } from './app.module';

async function bootstrap() {
    const app = await NestFactory.create<NestExpressApplication>(AppModule);
    await app.listen(3001);
}
bootstrap();
```
or alternatively:
```ts
import {} from '@nestjs/platform-express';
...
const app = await NestFactory.create(AppModule);
```

Ultimately my questions: is this a limitation of Denoify? Or is there a way to tell the generator to somehow import that dependency? Or would it be expected to have some `deps.deno.ts` file or something of the sort?

**Comment by garronej** [comment]
Hi @mmacdonaldOCP,  

Is the application open source?  
What's the error your getting? Runtime? Compile time?

**Comment by mattmacdonald-m42** [comment]
Hi @garronej 👋 

It's a runtime error, and specifically it's with NestJS framework as it's expecting the dependency I specified above to be listed in `node_modules`.

Specifically:
`[Nest] 14636  - 03/10/2023, 6:17:18 PM   ERROR [PackageLoader] No driver (HTTP) has been selected. In order to take advantage of the default driver, please, ensure to install the "@nestjs/platform-express" package ($ npm install @nestjs/platform-express).`

As far as the application it's just the initial output from creating a [NestJS](https://nestjs.com/) app. The code I posted above is the generated code in `main.ts`. Basically if you'd like a local repro you'd just run these steps:
```
npm i -g @nestjs/cli
nest new project-name
cd project-name
npm i @nestjs/platform-express
```
Then I just add Denoify as a dev dependency and run as-is in root project folder. After all that running Deno from root with:
```
deno run --allow-read --allow-env --allow-sys --allow-net deno_dist/main.ts
```
